### ISSUE TITLE
ENH:  scipy.spatial.ConvexHull for many different polyhedra in parallel

### ISSUE THREAD

**Issue by fzimmermann89** [scipy/scipy#20118]
### Is your feature request related to a problem? Please describe.

In MR Imaging, we need to calculate for long lists of 2D or 3D points the volume of the voronoi cells around each point.

The vornoi tesselation is really fast, leading to a lists of length 10k of 6-40 `vertices `for each point. 
 Using `dcf=[scipy.spatial.ConvexHull(v).volume for v in vertices]` is really slow.
Using a `ThreadPoolExecutor `seems to be limited by overhead /  non-nogil regions (?) and uses only 200% CPU on a 64 Core machine
Using a _ProcessPoolExecutor_ is also just a factor of 2-3 faster than the list version, most likely due to overhead.

### Describe the solution you'd like.

It would be nice to pass the vertices list to Convexhull, which then uses cython multihreading to call qhull.
Calls to .volume etc would then return lists as well.

For the interface, this would be a bit similar to how scipy.spatial.transform.Rotation also handles arrays of Rotations

### Describe alternatives you've considered.

We could also move this into a separate package if this does not fit into scipy.
Do you have any suggestions why this does not parallelize nicely with a ThreadPoolExecutor?
It looked like most of the logic should be in a nogil block and should run in parallel threads..

**Comment by tylerjereddy** [comment]
Can you clarify--you have a single Voronoi diagram and you want to calculate the area (2D) or volume (3D) of the Voronoi region around each point/generator in that diagram? So you're calculating the convex hull of each Voronoi region just so you can access the `volume` attribute? Or is something else going on here?

Since Voronoi regions are guaranteed to be convex, I'd already worry about efficiency if that's what you're doing, but maybe you mean more than one Voronoi region near each point?

**Comment by fzimmermann89** [comment]
Yes, this seems unnecessarily complex considering the voronoi cells will always be convex. 
So directly having a calculate_volume on Vornoi would be the best for us.

**Comment by tylerjereddy** [comment]
I was looking at this a bit today. Pauli noted that the convex hull of each region is indeed the recommended way: https://stackoverflow.com/a/19661317/2942522. It is even mentioned in the Qhull docs: http://www.qhull.org/html/qh-faq.htm#volume

Apparently there's a lib that can do it faster, referenced in the Qhull docs: https://www.netlib.org/voronoi/hull.html

Anyway, let me think about this a bit more. Pauli seemed to reach the same conclusion I did, the direct "volume of vor region" is not exposed directly in Qhull. We could probably provide something fairly fast under the hood though.

**Comment by tylerjereddy** [comment]
We might be able to lift the ideas from the netlib hull lib, assuming this is license compatible:

```
/* hull.c : "combinatorial" functions for hull computation */

/*
 * Ken Clarkson wrote this.  Copyright (c) 1995 by AT&T..
 * Permission to use, copy, modify, and distribute this software for any
 * purpose without fee is hereby granted, provided that this entire notice
 * is included in all copies of any software which is or includes a copy
 * or modification of this software and in all copies of the supporting
 * documentation for such software.
 * THIS SOFTWARE IS BEING PROVIDED "AS IS", WITHOUT ANY EXPRESS OR IMPLIED
 * WARRANTY.  IN PARTICULAR, NEITHER THE AUTHORS NOR AT&T MAKE ANY
 * REPRESENTATION OR WARRANTY OF ANY KIND CONCERNING THE MERCHANTABILITY
 * OF THIS SOFTWARE OR ITS FITNESS FOR ANY PARTICULAR PURPOSE.
 */
```

**Comment by rkern** [comment]
FWIW, this has the [same annoying documentation clause](https://github.com/scipy/unuran/pull/14) of the function that we just removed from our UNU.RAN fork.

**Comment by fzimmermann89** [comment]
@tylerjereddy Did you make any progress here or have any suggestions on how to get the volume of the 3D vornoi cels reasonably fast?

**Comment by tylerjereddy** [comment]
I haven't done much here yet, but I'll make a note to revisit--I think the use case is compelling.
